src/sysupdate/sysupdated.c

```diff
@@ -0,0 +1,1895 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "sd-bus.h"
+
+#include "build-path.h"
+#include "bus-get-properties.h"
+#include "bus-label.h"
+#include "bus-log-control-api.h"
+#include "bus-polkit.h"
+#include "common-signal.h"
+#include "discover-image.h"
+#include "env-util.h"
+#include "event-util.h"
+#include "escape.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "main-func.h"
+#include "memfd-util.h"
+#include "mkdir-label.h"
+#include "os-util.h"
+#include "process-util.h"
+#include "service-util.h"
+#include "signal-util.h"
+#include "socket-util.h"
+#include "string-table.h"
+
+typedef struct Manager {
+        sd_event *event;
+        sd_bus *bus;
+
+        Hashmap *targets;
+
+        uint64_t current_job_id;
+        Hashmap *jobs;
+
+        Hashmap *polkit_registry;
+
+        sd_event_source *notify_event;
+} Manager;
+
+typedef enum TargetClass {
+        /* These should try to match ImageClass from src/basic/os-util.h */
+        TARGET_MACHINE  = IMAGE_MACHINE,
+        TARGET_PORTABLE = IMAGE_PORTABLE,
+        TARGET_SYSEXT   = IMAGE_SYSEXT,
+        TARGET_CONFEXT  = IMAGE_CONFEXT,
+
+        /* sysupdate-specific classes */
+        TARGET_HOST,
+        TARGET_COMPONENT,
+
+        _TARGET_CLASS_MAX,
+        _TARGET_CLASS_INVALID = -EINVAL,
+} TargetClass;
+/* Let's ensure when the number of classes is updated things are updated here too */
+assert_cc(_IMAGE_CLASS_MAX == 4);
+
+typedef struct Target {
+        Manager *manager;
+
+        TargetClass class;
+        char *name;
+        char *path;
+        char *version;
+        char **appstream;
+
+        char *id;
+        ImageType image_type;
+        bool busy;
+} Target;
+
+typedef enum JobType {
+        JOB_LIST,
+        JOB_DESCRIBE,
+        JOB_CHECK_NEW,
+        JOB_UPDATE,
+        JOB_VACUUM,
+        _JOB_TYPE_MAX,
+        _JOB_TYPE_INVALID = -EINVAL,
+} JobType;
+
+typedef struct Job Job;
+
+typedef int (*JobComplete)(sd_bus_message *, const Job *, sd_json_variant *, sd_bus_error *);
```

Comment:
please give parameters useful names. I mean for some it doesn't matter much, but the 3rd arg is otherwise a bit too opaque